Filter root spans

stoeffel opened the pull request:
![filtering](https://user-images.githubusercontent.com/1217681/109524989-5d6a9c00-7ab1-11eb-849b-437ff4ba0e21.gif)

jwoudenberg-nri commented on nri-log-explorer/package.yaml:
💭 This seems to pull in a big dependency `monoid-subclasses`. Might be worth seeing if we can drop it at some point.

jwoudenberg-nri commented on nri-log-explorer/src/Main.hs:
❔ Given that `ListWidget.List` is backed by a `Vector`, I wonder if keeping the original list in a `Vector` as well will mean we need to do fewer conversions when updating a filter.

jwoudenberg-nri commented on nri-log-explorer/src/Main.hs:
😻

jwoudenberg-nri commented on nri-log-explorer/src/Main.hs:
🎨 Maybe rename `Filtering` to `EditingFilter`?

jwoudenberg-nri commented on nri-log-explorer/src/Main.hs:
🎨 Maybe `StopEditingFilter`?

jwoudenberg-nri commented on nri-log-explorer/src/Main.hs:
😻

stoeffel commented on nri-log-explorer/src/Main.hs:
done

stoeffel commented:
@jwoudenberg-nri I've addressed all your comments.
I think there are no alternative packages for fuzzy that I can find on hackage :-(
I think our options are drop fuzzy filtering (it's so nice though) or write/fork our own fuzzy. I'm happy to fork, but I think that's out of scope for this PR, except if you think it's blocking it. Thoughts?

jwoudenberg-nri reviewed (approved):
This is great, thank you so much!

![](https://media.giphy.com/media/4QFAV6hISzSjPyaWS2/giphy.gif)

jwoudenberg-nri commented:
> @jwoudenberg-nri I've addressed all your comments.

Sorry, was still reviewing so I added a few more!

> I think there are no alternative packages for fuzzy that I can find on hackage :-(
> I think our options are drop fuzzy filtering (it's so nice though) or write/fork our own fuzzy. I'm happy to fork, but I think that's out of scope for this PR, except if you think it's blocking it. Thoughts?

I definitely want to keep fuzzy searching. I wonder if creating our own function for matching would be a lot of work, but definitely agree it's out of scope for this PR. Just a thought that occurred to me while reviewing, nothing I feel we need to act on now!

stoeffel commented on nri-log-explorer/src/Main.hs:
We do get live updates as we type.
Might make it a little simpler though. will check

stoeffel commented on nri-log-explorer/src/Main.hs:
I think we can't call `handleEditorEvent` in "update" though right?

stoeffel commented on nri-log-explorer/src/Main.hs:
Actually, I think it makes it harder. Because we also want to track previous filters for the esc behaviour.

omnibs commented on nri-prelude/src/Test/Internal.hs:
Do you know when we can have a test w/o a corresponding `SrcLoc`?

omnibs commented on nri-prelude/src/Test/Internal.hs:
Explain what this does?

omnibs commented on nri-prelude/src/Test/Internal.hs:
We discussed it and it seems it might not happen in practice. Maybe we should remove the `Maybe` from `loc`.

omnibs reviewed (approved):
This looks good, but I made significant changes, so I'd love a second look from @stoeffel before merging
